src/Bicep.Cli/Services/TemplateEvaluatorService.cs

```diff
@@ -0,0 +1,208 @@
+                configuration = configBuilder(configuration);
+            }
+
+            return EvaluateTemplate(templateJtoken, parametersJToken, configuration);
+        }
+
+        private static JToken EvaluateTemplate(JToken? templateJtoken, JToken? parametersJToken, EvaluationConfiguration config)
+        {
+            templateJtoken = templateJtoken ?? throw new ArgumentNullException(nameof(templateJtoken));
+
+            var deploymentScope = GetDeploymentScope(templateJtoken["$schema"]!.ToString());
+
+            var metadata = new InsensitiveDictionary<JToken>();
+
+            try
+            {
+                var template = TemplateEngine.ParseTemplate(templateJtoken.ToString());
+                var parameters = ParseParametersFile(parametersJToken);
+                
+                TemplateEngine.ValidateTemplate(template, "2020-10-01", deploymentScope);
+                TemplateEngine.ParameterizeTemplate(template, new InsensitiveDictionary<JToken>(parameters), metadata, null, new InsensitiveDictionary<JToken>());
+
+                TemplateEngine.ProcessTemplateLanguageExpressions(config.ManagementGroup, config.SubscriptionId, config.ResourceGroup, template, "2020-10-01", null);
+
+                ProcessTemplateLanguageExpressions(template, config, deploymentScope);
+
+                TemplateEngine.ValidateProcessedTemplate(template, "2020-10-01", deploymentScope);
+
+                return template.ToJToken();
+            }
+            catch (Exception exception)
+            {
+                throw new InvalidOperationException(
+                    $"Evaluating template failed: {exception.Message}." +
+                    $"\nTemplate file: {templateJtoken}" +
+                    (parametersJToken is null ? "" : $"\nParameters file: {parametersJToken}"),
+                    exception);
+            }
+        }
+
+        private static ImmutableDictionary<string, JToken> ParseParametersFile(JToken? parametersJToken)
+        {
+            if (parametersJToken is null)
+            {
+                return ImmutableDictionary<string, JToken>.Empty;
+            }
+
+            return parametersJToken.Cast<JProperty>().ToImmutableDictionary(x => x.Name, x => x.Value!);
+        }
+        private static TemplateDeploymentScope GetDeploymentScope(string templateSchema)
+            => templateSchema switch
+            {
+                "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#" => TemplateDeploymentScope.Tenant,
+                "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#" => TemplateDeploymentScope.ManagementGroup,
+                "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#" => TemplateDeploymentScope.Subscription,
+                "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#" => TemplateDeploymentScope.ResourceGroup,
```

Comment:
The API versions for the template schemas may change, so it would be better to use regex matching instead. You may copy the regex from https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/_bicep.py#L333